### PR TITLE
chore(candy-machine): bump mpl-token-metadata version and fix new cli…

### PR DESCRIPTION
**Pull request by samuelvanderwaal**
To support sized collections in Sugar, we need to rely on `mpl-token-metadata` 1.3+ which requires `mpl-candy-machine` to bump its version of that crate and import the `TokenMetadataAccount` trait.

This PR also fixes the new clippy lint warnings introduced in Rust v1.63.0.

This should result in a minor bump in the`mpl-candy-machine` crate version, IMO, to v4.1.0.

### CODE REVIEW

**Review by samuelvanderwaal — COMMENTED** [review]
version minor

**Review by stranzhay — APPROVED** [review]

**Comment by stegaBOB** [comment]
Why did so much change?

File: candy-machine/js/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@metaplex-foundation/mpl-candy-machine",
-  "version": "4.3.0",
+  "version": "4.4.0",
   "programVersion": "4.0.2",
```
Comment (stegaBOB) [inline candy-machine/js/package.json:4]: Should this be 4.1?

**Review by stegaBOB — CHANGES_REQUESTED** [review]

File: candy-machine/js/package.json (hunk 1 above)
Comment (samuelvanderwaal) [reply candy-machine/js/package.json:4]: No, the JS SDKs are versioned independently.

**Comment by samuelvanderwaal** [comment]
> Why did so much change?

About half of the Rust changes are new clippy lints released in new version of clippy that came with Rust v1.63.0.

Is there a specific change that is concerning? We can do a huddle review if you wan to walk through the changes together.

**Comment by stegaBOB** [comment]
I was talking about the lockfile. Did something else get bumped or something?

**Review by stegaBOB — APPROVED** [review]

**Pull request merged**